vowpalwabbit/core/src/array_parameters_sparse.cc

```diff
@@ -26,6 +26,21 @@ VW::weight* VW::sparse_parameters::get_or_default_and_get(size_t i) const
   return iter->second.get();
 }
 
+VW::weight* VW::sparse_parameters::get_impl(size_t i) const
+{
+  static VW::weight default_value = 0.0f;
+
+  uint64_t index = i & _weight_mask;
+  auto iter = _map.find(index);
```

Comment:
bloom filter might make things faster here if find() is slow

Reply:
Bloom filter seems to not be deterministic which would probably be an issue. find() should have an average O(1) lookup but its slow since its looking for elements not in the map, which will always be O(n)